crates/rethnet_evm/benches/state_history.rs

```diff
@@ -177,6 +180,66 @@ fn bench_set_block_context_to_middle_layer(c: &mut Criterion) {
     );
 }
 
+fn bench_set_block_context_to_block_number(
+    c: &mut Criterion,
+    group_name: &str,
+    fork_block_number: U256,
+    context_block_number: U256,
+) {
+    let mut group = c.benchmark_group(group_name);
+    for accounts_per_checkpoint in CHECKPOINT_SCALES.iter() {
+        for number_of_accounts in ADDRESS_SCALES.iter() {
+            for number_of_snapshots in SNAPSHOT_SCALES.iter() {
```

Comment:
This is not currently doing anything with `number_of_snapshots`

Reply:
I hope you don't mind, but I took the liberty of re-basing this PR on the `StateHistory` one, since some stuff over there changed significantly, and since this PR isn't all THAT much change. In the commits I just pushed, you'll see that I'm using `number_of_snapshots` there now.